src/input/mod.rs

```diff
@@ -3449,6 +3449,19 @@ pub fn apply_libinput_settings(config: &niri_config::Input, device: &mut input::
         } else {
             input::SendEventsMode::ENABLED
         });
+
+        #[rustfmt::skip]
+        const MATRIX: [f32; 6] = [
+            1., 0., 0.,
+            0., 1., 0.,
+        ];
```

Comment:
Instead of hardcoding the default matrix, you should get it from [`libinput_device_config_calibration_get_default_matrix()`](https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga3d9f1b9be10e804e170c4ea455bd1f1b) since it can be overridden in udev.

Reply:
I think the rust binding for this returns an option, how should we deal with that?